### PR TITLE
Scan with arm vs scan with base

### PR DESCRIPTION
**Pull request by KavitShah1998**

### CODE REVIEW

File: spot_rl_experiments/spot_rl/utils/img_publishers.py [hunk 1]
```diff
@@ -622,7 +622,7 @@ def _publish(self):
                     print(f"Affordance Point is NaN for {class_label}, skipping")
                     continue
             except Exception as e:
-                print(f"Issue of predicting location: {e}")
+                print(f"Issue of predicting location for {class_label} : {e}")
```
Comment (KavitShah1998) [inline spot_rl_experiments/spot_rl/utils/img_publishers.py:625]: @jimmytyyang maybe we can disable SpotOpenVocObjectDetectorPublisher fully.

This is only being used by Nexus, and it doesn't matter to us in vizusalization there about how many objects are being detected and sent for target object highlight.

Lmk what you think

File: spot_rl_experiments/spot_rl/utils/img_publishers.py [hunk 2]
```diff
@@ -686,8 +686,8 @@ def inference(self, hand_rgb, timestamp, stopwatch):
         # Add new classes here to the model
         # We decide to hardcode these classes first as this will be more robust
         # and gives us the way to control the detection
-        # multi_classes = [["ball", "cup", "table", "cabinet", "chair", "sofa"]]
-        multi_classes = rospy.get_param("/multi_class_object_target").split(",")
+        multi_classes = ["ball", "cup"]
+        # multi_classes = rospy.get_param("/multi_class_object_target").split(",")
```
Comment (jimmytyyang) [inline spot_rl_experiments/spot_rl/utils/img_publishers.py:690]: I would like to be used with `rospy.get_param("/multi_class_object_target")` to set the object name

File: spot_rl_experiments/spot_rl/utils/img_publishers.py [hunk 3]
```diff
@@ -640,8 +640,8 @@ def _publish(self):
             )
 
         # publish data
-        self.publish_new_detection(";".join(object_info))
-        self.publish_viz_img(viz_img, header)
+        # self.publish_new_detection(";".join(object_info))
```
Comment (jimmytyyang) [inline spot_rl_experiments/spot_rl/utils/img_publishers.py:643]: why remove this?

File: spot_rl_experiments/spot_rl/utils/img_publishers.py (hunk 1 above)
Comment (jimmytyyang) [reply spot_rl_experiments/spot_rl/utils/img_publishers.py:625]: this is used for target object highlight -- let's chat

File: spot_rl_experiments/spot_rl/utils/heuristic_nav.py [hunk 4]
```diff
@@ -157,12 +175,34 @@ def get_arguments_for_image_search(spot: Spot):
...
+            "vision", "hand_color_image_sensor", imgs[0].shot.transforms_snapshot
+        )
+        body_T_hand = spot.get_magnum_Matrix4_spot_a_T_b(
+            "body", "hand_color_image_sensor", imgs[0].shot.transforms_snapshot
+        )
+    except Exception as e:
+        print(e)
+        breakpoint()
+
+    if any(
```
Comment (jimmytyyang) [inline spot_rl_experiments/spot_rl/utils/heuristic_nav.py:192]: why this?

File: spot_rl_experiments/spot_rl/models/owlvit.py [hunk 5]
```diff
@@ -127,9 +125,6 @@ def run_inference_and_return_img(
             results = self.processor.post_process_object_detection(
                 outputs=outputs, target_sizes=target_sizes
             )
-        # img = img.to('cpu')
```
Comment (jimmytyyang) [inline spot_rl_experiments/spot_rl/models/owlvit.py:130]: why this?

**Review by jimmytyyang — COMMENTED** [review]
left some comments, thanks a lot!

**Comment by KavitShah1998** [comment]
@jimmytyyang  can we close this PR since we merged the important bits from @tusharsangam 's branch?

**Comment by jimmytyyang** [comment]
thank you @KavitShah1998 ! agree with you, thank you!

**Comment by KavitShah1998** [comment]
Closing this PR. The important code from this PR was merged via this [PR](https://github.com/facebookresearch/spot-sim2real/pull/244)

**Pull request closed without merging**